lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+    "* In general, this means an \"Occam's Razor\"-like penalty for complexity should be involved (to avoid focusing on models that \"over-fit\" the data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "In our example, we might add a quadratic term to the model: $y = b + m x + q x^2$. How do we quantify the improvement?\n",
+    "\n",
+    "<table><tr>\n",
+    "<td><img src=\"graphics/modelcheck-quadratic.png\" width=80%></td>\n",
+    "<td><img src=\"graphics/modelcheck-quadratic-discrepancy.png\" width=80%></td>\n",
+    "</tr></table>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "The gold standard for testing predictive accuracy is to _get more data_.\n",
+    "\n",
+    "Short of that, the best option is **cross-validation**: fitting a model on many random subsets of the data and seeing how well it describes the complementary \"out of sample\" subsets.\n",
+    "\n",
+    "> This method is ubiquitous in machine learning, where accurate out-of-sample prediction is usually the goal."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Short of exhaustive cross-validation, a number of **information criteria** exist that (asymptotically) relate to generalized predictive accuracy.\n",
+    "\n",
+    "These have the advantage of being relatively quick to calculate from the results of a fit - either an MLE or a set of posterior samples - and include a penalty for models with greater freedom."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Some information criteria:\n",
+    "* Akaike information criterion (AIC)\n",
+    "* Deviance information criterion (DIC)\n",
+    "* Watanabe-Akaike information criterion (WAIC)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "The DIC has the advantage of being compatible with Bayesian analysis (unlike AIC), and not requiring the data to be cleanly separable into conditionally independent subsets (unlike WAIC).\n",
+    "\n",
+    "$\\mathrm{DIC} = \\langle D(\\theta) \\rangle + p_D; \\quad p_D = \\langle D(\\theta) \\rangle - D(\\langle\\theta\\rangle)$\n",
+    "\n",
+    "where $D(\\theta)=-2\\log P(\\mathrm{data}|\\theta)$ and averages $\\langle\\rangle$ are over the posterior.\n",
+    "\n",
+    "$p_D$ is an _effective number of free parameters_, i.e. the number of parameters primarily constrained by the data rather than by their priors.\n",
+    "\n",
+    "The DIC thus doesn't necessarily count nuisance parameters used to marginalize out systematics as added \"complexity\"."
```

Comment:
Given the discussion in Hierarchical Modeling, maybe this should say "unconstrained nuisance parameters"

Reply:
Yes, good